openhtf/plugs/usb/__init__.py

```diff
@@ -36,56 +38,153 @@ def MyPhase(test, adb):
 conf.Declare('usb_server', 'USB Server IP/Hostname')
 conf.Declare('usb_server_port', 'USB Server Port',
              default_value=10000)
-
 conf.Declare('libusb_rsa_key', 'A private key file for use by libusb auth.')
+conf.Declare('device_type', 'android, fastboot or general usb type')
+conf.Declare('usb_hub', 'local or Ethersyncy')
+
+_LOG = logging.getLogger(__name__)
+
+class GeneralUsbAttributeError(Exception):
+  """Indicates the usb class/subclass/protocol info was invalid."""
+
+class general_usb(object):
+  def __init__(self, usb_class=None, usb_subclass=None, usb_protocol=None):
+    self.usb_class = usb_class
+    self.usb_subclass = usb_subclass
+    self.usb_protocol = usb_protocol
+
+  @property
+  def CLASS(self):
+    return self.usb_class
+  @property
+  def SUBCLASS(self):
+    return self.usb_subclass
+  @property
+  def PROTOCOL(self):
+    return self.usb_protocol
+
+""" Config file example for device at Ethersync
+device_type: android (fastboot, general)
+usb_hub: 
+  Ethersync:
+   unit_name: EtherSyncca9167
```

Comment:
Unit name can be derived from MAC address, so we shouldn't need a separate key for it.